### PR TITLE
PopScope example improvements

### PR DESCRIPTION
**Pull request by justinmc**
Updates two PopScope examples to a more clear usage of showDialog.

This came up in https://github.com/flutter/flutter/issues/138614, where many users were confused about how to use a back confirmation dialog with PopScope. Two examples of this already exist, but now they are cleaner and follow more typical navigation patterns, so users should recognize their own code in them more easily.

### CODE REVIEW

File: examples/api/lib/widgets/pop_scope/pop_scope.0.dart [hunk 1]
```diff
@@ -62,8 +62,12 @@ class _PageTwo extends StatefulWidget {
 }
 
 class _PageTwoState extends State<_PageTwo> {
-  void _showBackDialog() {
-    showDialog<void>(
+  /// Shows a dialog and resolves to true when the user has indicated that they
+  /// want to pop.
+  ///
+  /// A null indicates a desire not to pop.
```
Comment (LongCatIsLooong) [inline examples/api/lib/widgets/pop_scope/pop_scope.0.dart:68]: Where's null introduced?

**Review by LongCatIsLooong — COMMENTED** [review]
I looked at https://main-api.flutter.dev/flutter/widgets/PopScope-class.html the example description only says "This sample demonstrates how to use this widget to handle nested navigation in a bottom navigation bar.". 

If I was looking for information on how to use an async function in PopScope I would probably turn away upon reading that description since it doesn't sound that interesting.

File: examples/api/lib/widgets/pop_scope/pop_scope.0.dart (hunk 1 above)
Comment (justinmc) [reply examples/api/lib/widgets/pop_scope/pop_scope.0.dart:68]: showDialog can return a Future that resolves to null. Dismissing the dialog (such as by tapping the barrier) will resolve to null. In those cases, the user doesn't want to leave the page.

I'll change that to "A return value of null indicates a desire not to pop, such as when the user has dismissed the modal without tapping a button."

**Comment by justinmc** [comment]
@LongCatIsLooong Good catch, that example description is actually totally wrong. I must have copied it from one of the navigator_pop_handler examples by mistake. I've updated it.

**Review by LongCatIsLooong — APPROVED** [review]
LGTM

**Pull request merged**